lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx

```diff
@@ -222,19 +245,80 @@ class PointConditionalDistributionVBoundEvaluation : public EvaluationImplementa
 };
 
 
+void PointConditionalDistribution::dispatchConditioning(const Collection<Distribution> & distributions, Distribution & simplified) const
+{
+  Collection<Indices> conditioningIndicesBlocks(distributions.getSize());
+  Collection<Point> conditioningValuesBlocks(distributions.getSize());
+  // Sort both the conditioning indices and values in increasing order
+  const UnsignedInteger conditioningSize = conditioningIndices_.getSize();
+  Collection< std::pair<UnsignedInteger, Scalar> > sortedConditioningPairs(conditioningSize);
+  for (UnsignedInteger i = 0; i < conditioningSize; ++i)
+    sortedConditioningPairs[i] = std::pair<UnsignedInteger, Scalar>(conditioningIndices_[i], conditioningValues_[i]);
+  std::sort(sortedConditioningPairs.begin(), sortedConditioningPairs.end());
+  UnsignedInteger beginBlock = 0;
+  UnsignedInteger endBlock = distributions[0].getDimension();
+  UnsignedInteger blockCounter = 0;
+  for (UnsignedInteger conditioningCounter = 0; conditioningCounter < conditioningSize; ++conditioningCounter)
+  {
+    const UnsignedInteger conditioningIndex = sortedConditioningPairs[conditioningCounter].first;
+    // Find the block the conditioning index belongs to
+    while (conditioningIndex >= endBlock)
+    {
+      ++ blockCounter;
+      // Should never go there
+      if (blockCounter == distributions.getSize())
+        break;
+      beginBlock = endBlock;
+      endBlock += distributions[blockCounter].getDimension();
+    }
+    conditioningIndicesBlocks[blockCounter].add(conditioningIndex - beginBlock);
+    conditioningValuesBlocks[blockCounter].add(sortedConditioningPairs[conditioningCounter].second);
+  } // for conditioningCounter
+  // Now, decide what to do for each block
+  Collection<Distribution> newBlocks;
+  for (UnsignedInteger i = 0; i < distributions.getSize(); ++i)
+  {
+    // If the block is fully conditioned, skip it
+    if (conditioningIndicesBlocks[i].getSize() == distributions[i].getDimension())
+      continue;
+    // If the block is not conditioned
+    if (conditioningIndicesBlocks[i].getSize() == 0)
+    {
+      newBlocks.add(distributions[i]);
+      continue;
+    }
+    const PointConditionalDistribution conditionalBlock(distributions[i], conditioningIndicesBlocks[i], conditioningValuesBlocks[i]);
+    if (conditionalBlock.useSimplifiedVersion_)
```

Comment:
As discussed during the technical meeting, this line and the lines below can be replaced by
```
newBlocks.add(conditionalBlock.getSimplifiedVersion());
```
as the simplified version is the PointConditionalDistribution if no simplification occurred.

Reply:
I missed that